packages/ember-views/tests/views/view/child_views_test.js

```diff
@@ -68,3 +69,41 @@ QUnit.test('should not duplicate childViews when rerendering', function() {
     outerView.destroy();
   });
 });
+
+QUnit.test('should remove childViews on destroy', function() {
+  var outerView = EmberView.extend({
+    component: 'my-thing',
+    value: false,
+    container: {
+      lookup() {
+        return {
+          componentFor(tagName, container) {
+            return Component.extend({
+              destroy() {
```

Comment:
We should probably add an assertion here, to confirm that `destroy` is called...